lib/ui/painting.dart

```diff
@@ -2350,22 +2341,27 @@ class _PathMeasure extends NativeFieldWrapperClass2 {
       );
     }
   }
-  Float32List _getPosTan(double distance) native 'PathMeasure_getPosTan';
+  Float32List _getPosTan(int contourIndex, double distance) native 'PathMeasure_getPosTan';
 
-  Path extractPath(double start, double end, {bool startWithMoveTo: true}) native 'PathMeasure_getSegment';
+  Path extractPath(int contourIndex, double start, double end, {bool startWithMoveTo: true}) {
+    assert(contourIndex <= currentContourIndex, 'Iterator must be advanced before index $contourIndex can be used.');
+    return _extractPath(contourIndex, start, end, startWithMoveTo: startWithMoveTo);
+  }
+  Path _extractPath(int contourIndex, double start, double end, {bool startWithMoveTo: true}) native 'PathMeasure_getSegment';
 
-  bool get isClosed native 'PathMeasure_isClosed';
+  bool isClosed(int contourIndex) {
+    assert(contourIndex <= currentContourIndex, 'Iterator must be advanced before index $contourIndex can be used.');
+    return _isClosed(contourIndex);
+  }
+  bool _isClosed(int contourIndex) native 'PathMeasure_isClosed';
 
   // Move to the next contour in the path.
   //
   // A path can have a next contour if [Path.moveTo] was called after drawing began.
   // Return true if one exists, or false.
   //
-  // This is not exactly congruent with a regular [Iterator.moveNext].
-  // Typically, [Iterator.moveNext] should be called before accessing the
-  // [Iterator.current]. In this case, the [PathMetric] is valid before
-  // calling `_moveNext` - `_moveNext` should be called after the first
-  // iteration is done instead of before.
+  // Before Skia introduced an SkPathContourMeasureIter, this didn't work like
+  // a normal iterator.  Now it does.
```

Comment:
nit: avoid talking about the history of the code in the comments. it doesn't really matter to the person reading tho code.